log/src/io/pedestal/log.clj

```diff
@@ -109,6 +109,11 @@
     ([t body] nil)
     ([t body throwable] nil)))
 
+(def ^:dynamic ^{:added "0.5.4"} *mdc*
```

Comment:
`MDC` is already well known as _Mapped Diagnostic Context_ in logging implementations. Using a similar abbreviation here is confusing, particularly when since the semantics differ.

Reply:
Calling it the message diagnostic context may have just been a slip of the tongue.

It's possible to use the MDC from Logback BUT that has two problems:
- The added data is outside the EDN map emitted by pedestal.log
- Coordinating the per-thread MDC data in Logback against multiple threads (especially with core.async) is problematic

The solutions we've used for the MDC data usually ended up involving some leaked data; the MDC would be updated in a thread and that data might persist after the logging on that thread completed. So you had to be very careful to initialize all the potential keys of the MDC early in each request pipeline, which is quite haphazard.

We don't currently use pedestal.log in our application production code, we use our own wrappers around Logback. This is for some open source code I'm working on, but if it goes through I can start looking into moving our production code to pedestal.log.